pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -0,0 +1,213 @@
+package configurableinterpreter
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/engine"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/engine/lua"
```

Comment:
I suggest adding this package to `hack/.import-aliases`.

Reply:
Do it in later.